core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeAstBuilder.scala

```diff
@@ -63,4 +63,14 @@ class SnowflakeAstBuilder extends SnowflakeParserBaseVisitor[ir.LogicalPlan] wit
     case c if c.queryStatement() != null => c.queryStatement().accept(this)
     case c => c.accept(dmlBuilder)
   }
+
+  override def visitSnowSqlCommand(ctx: SnowSqlCommandContext): ir.UnresolvedCommand = {
+    val input = ctx.getText.trim
+    ir.UnresolvedCommand(if (input.endsWith(";")) { /* Removing ; as it will get appended in generator */
```

Comment:
Don't trim or check for ';'. Then in the generator do not add anything to the text of the command - preserve as it was in the input.

Reply:
done

Reply:
Hi @jimidle, in the generate method within LogicalPlanGenerator.scala, there's a case statement:
case b: ir.Batch => b.children.map(generate(ctx, _)).mkString("", ";\n", ";")

This appends a ";" at the end of the batch.
If I don't remove the ";" in visitSnowSqlCommand, we'll end up with two ";" at the end of the generate method.

Reply:
I guess it can be left as is.